presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -257,7 +266,20 @@ private Date getDate(int columnIndex, DateTimeZone localTimeZone)
         }
 
         try {
-            return new Date(DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value)));
+            long millis = DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value));
+            if (millis >= START_OF_MODERN_ERA) {
+                return new Date(millis);
+            }
+
+            // The chronology used by default by Joda is not historically accurate for dates preceding the introduction of the Gregorian calendar
+            // and is not consistent with java.sql.Date (the same millisecond value represents a different year/month/day) before the 20th century.
+            // For such dates we are falling back to using the more expensive GregorianCalendar; note that Joda also has a chronology that works for
+            // older dates, but it uses a slightly different algorithm and still yields results that are not compatible with java.sql.Date.
+            LocalDate localDate = DATE_FORMATTER.parseLocalDate(String.valueOf(value));
+            GregorianCalendar calendar = new GregorianCalendar(localDate.getYear(), localDate.getMonthOfYear() - 1, localDate.getDayOfMonth());
```

Comment:
This can be declared `Calendar` since we don't use any methods specific to `GregorianCalendar`.